src/gui/app.cpp

```diff
@@ -116,10 +116,16 @@ void App::showUi() noexcept
 		win->show();
 	}
 #else
+	setQuitOnLastWindowClosed(false);
 	NeovimQt::MainWindow *win = new NeovimQt::MainWindow(m_connector.get(), opts);
 
-	QObject::connect(instance(), SIGNAL(openFilesTriggered(const QList<QUrl>)),
+	// delete the main window when closed to emit `destroyed()` signal.
```

Comment:
nitpick/optional: You can clean up the diff by keeping the line below on its own at the top. Preserves better VCS history.

`QObject::connect(instance(), SIGNAL(openFilesTriggered(const QList<QUrl>)),`